src/dags/bbga.py

```diff
@@ -0,0 +1,245 @@
+import csv
+import itertools
+import shutil
+import tempfile
+from csv import Dialect
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from typing import Dict, Tuple, TypedDict
+
+from airflow import DAG
+from airflow.models import Variable
+from airflow.operators.postgres_operator import PostgresOperator
+from airflow.operators.python_operator import PythonOperator
+from common import DATAPUNT_ENVIRONMENT, MessageOperator, default_args, slack_webhook_token
+from http_fetch_operator import HttpFetchOperator
+from pgcomparator_cdc_operator import PgComparatorCDCOperator
+from postgres_create_tables_like_operator import PostgresCreateTablesLikeOperator
+from postgres_insert_csv_operator import FileTable, PostgresInsertCsvOperator
+from sqlalchemy_create_object_operator import SqlAlchemyCreateObjectOperator
+
+FileStem = str
+UrlPath = str
+
+DAG_ID = "bbga"
+TMP_DIR_BASE = Path("/tmp")
+TMP_TABLE_PREFIX = "tmp_"
+VARS = Variable.get(DAG_ID, deserialize_json=True)
+data_endpoints: Dict[FileStem, UrlPath] = VARS["data_endpoints"]
+table_mappings: Dict[FileStem, str] = VARS["table_mappings"]
+
+assert set(data_endpoints.keys()) == set(
+    table_mappings.keys()
+), "Both mappings should have the same set of keys."
+
+
+class bbga_dialect(Dialect):
+    """Describe the usual properties of BBGA CSV files."""
+
+    delimiter = ";"
+    quotechar = '"'
+    doublequote = True
+    skipinitialspace = False
+    lineterminator = "\n"
+    quoting = csv.QUOTE_NONE
+
+
+ColName = str
+
+
+class CsvDef(TypedDict):
+    dialect: Dialect
+    fieldnames: Tuple[ColName, ...]
+    composite: Dict[ColName, Tuple[ColName, ...]]
+
+
+CSV_DEFINITIONS: Dict[FileStem, CsvDef] = {
```

Comment:
Would it be possible to derive this from the bbga amsterdam schema?

Reply:
Geen idee. De naamgeving van de kolommen wel, maar de volgorde mogelijkerwijs niet. Dat de volgorde van belang is hier af te leiden uit het feit dat het in een datastructuur met ordening zit; een`tuple`. In het schema zijn het echter properties. Hebben properties een vaste volgorde? Ik weet het niet zeker, maar dacht van niet.